### PR TITLE
chore(tests): regression fix for `master-b352814976`

### PR DESCRIPTION
**Pull request by canonbrother**
#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:

- [x] deps: ethers & hardhat installed for evm stuff
- [x] container: add `-ethrpcbind` to support eth rpc call + add `RUST_LOG=debug`
- [x] CI
- [x] enhance `checkIfEvmTx`
- [x] fix: TD test (dftx composer, rpc & rawtx)
- [x] func: add `ListUnspentQueryOptions` to support provider `listunspent` method
- [x] fix: UpdateMN test
- [x] lint: ts-ignore buggy AbortController.signal type check

### CODE REVIEW

**Comment by canonbrother** [comment]
due to that has updatemn bug on beta-12.. so switch to more later version `master-4430349b5f`

**Comment by canonbrother** [comment]
for own note at https://github.com/BirthdayResearch/jellyfishsdk/pull/2149/commits/eb3ee72608b979bf8023b596a24bcdc0599951f1
skipping few flaky and unexpected behaviour test.. (they were passed previously)
will fix on later fixed version

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 1]
```diff
@@ -527,17 +527,18 @@ describe('TransferDomain', () => {
       await expect(promise).rejects.toThrow('amount 90000.00000000 is less than 999999.00000000')
     })
 
-    it('(evm -> dvm) should fail if insufficient balance', async () => {
+    // TODO(canonbrother): check again here after pull the later version
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:530]: still need to check this? @canonbrother

Comment (canonbrother) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:530]: making a smaller scope with this specific fix on another PR is preferable.. we need a checkpoint first

Comment (canonbrother) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:530]: next check will be on `HEAD-53859f66f9`, the rc pre-release https://github.com/DeFiCh/ain/pull/2485
why we do a checkpoint here.. as this `master-b352814976` is the most stable version that i found while compare others ain against Jellyfish
we wouldn't know will the later regression cause other non-related test failed and doing kick and roll CI running or a real issue on evm which take days to get a release which's killing productivity on Jellyfish side

maybe take this as jellyfish evm compatible PR 🤔

Comment (pierregee) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:530]: got it

**Review by lykalabrada — APPROVED** [review]

**Review by pierregee — APPROVED** [review]

File: packages/jellyfish-transaction-builder/src/provider.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import BigNumber from 'bignumber.js'
 import { EllipticPair } from '@defichain/jellyfish-crypto'
 import { Vout } from '@defichain/jellyfish-transaction'
+import { ListUnspentQueryOptions } from '@defichain/jellyfish-api-core/dist/category/wallet'
```
Comment (fuxingloh) [inline packages/jellyfish-transaction-builder/src/provider.ts:4]: should not use `/dist`

**Review by fuxingloh — COMMENTED** [review]
I briefly skimmed through it, generally lgtm but the /dist might cause downstream usage error, should fix it

**Comment by fuxingloh** [comment]
Will merge first

File: packages/jellyfish-transaction-builder/src/provider.ts (hunk 2 above)
Comment (canonbrother) [reply packages/jellyfish-transaction-builder/src/provider.ts:4]: fixed https://github.com/BirthdayResearch/jellyfishsdk/pull/2156/commits/56612a2ff6bcd32af15ff96080b4caf8f678cdbd

**Pull request merged**